### ISSUE TITLE
[UX2.0] SNMP feature parcel fails when trap server is configured

### ISSUE THREAD

**Issue by tzarski0** [CiscoDevNet/terraform-provider-sdwan#205]
- When SNMPv2 is used, in the "target" block, the "userLabel" parameter should be present and "user" parameter should not be present.
- When SNMPv3 is used, in the "target" block, the "user" parameter should be present and "userLabel" parameter should not be present.

Right now when you configure any SNMP version, the provider puts both "user" and "userLabel" under the "target" which causes push to fail:
```
│ Error: Client Error
│ 
│   with module.sdwan.sdwan_system_snmp_profile_parcel.system_snmp_profile_parcel["system1-snmp"],
│   on ../terraform-sdwan-nac-sdwan/sdwan_profile_parcels.tf line 365, in resource "sdwan_system_snmp_profile_parcel" "system_snmp_profile_parcel":
│  365: resource "sdwan_system_snmp_profile_parcel" "system_snmp_profile_parcel" {
│ 
│ Failed to configure object (POST), got error: HTTP Request failed: StatusCode 400, {"error":{"message":"Invalid Json
│ Payload Input","code":"SCHVALID0001","details":"{\"Validation Errors\":{\"Invalid Format
│ Attributes\":[\"data.target[0].user\"]}}","type":"error"}}
```

**Comment by tzarski0** [comment]
One more problem for SNMPv2 resource:
```
resource "sdwan_system_snmp_profile_parcel" "system_snmp_profile_parcel" {
    communities                 = [
        {
            authorization = "read-write"
            name          = "test123"
            user_label    = "test"
            view          = "view_all"
         },
     ]
    contact_person_variable     = "{{snmp_contact}}"
    description                 = "basic snmpv2"
    feature_profile_id          = "76b21a53-b780-4e87-9652-71447378eb57"
    id                          = (known after apply)
    location_of_device_variable = "{{snmp_location}}"
    name                        = "snmpv2"
    shutdown_variable           = "{{snmp_shutdown}}"
    trap_target_servers         = [
        {
            ip                        = "172.16.0.11"
            port                      = 514
            source_interface_variable = "{{snmp_trap_source_interface}}"
            user_label                = "test"
            vpn_id                    = 511
         },
     ]
    version                     = (known after apply)
    views                       = [
        {
            name = "view_all"
            oids = [
                {
                    id = "1.3"
                 },
                {
                    exclude = true
                    id      = "1.3.5"
                 },
             ]
         },
     ]
 }
```

The TF provider configured resource without user and group keys (this causes UI not to display it correctly):
<img width="436" alt="Screenshot 2024-04-25 at 15 16 58" src="https://github.com/CiscoDevNet/terraform-provider-sdwan/assets/92273798/0a252f9e-ee65-411a-a33a-64bf1b1901bc">

However what it should do is it should push user and group as empty list:
<img width="370" alt="Screenshot 2024-04-25 at 15 16 11" src="https://github.com/CiscoDevNet/terraform-provider-sdwan/assets/92273798/966283db-0716-4024-b8df-70f0de512a86">

Looks like this causes problem as SNMPv2 configuration pushed by TF is detected as SNMPv3 in Manager UI.

**Comment by tzarski0** [comment]
For SNMPv3:
```
   resource "sdwan_system_snmp_profile_parcel" "system_snmp_profile_parcel" {
       contact_person_variable     = "{{snmp_contact}}"
       description                 = "basic snmpv3"
       feature_profile_id          = "<replace-id>"
       groups                      = [
           {
               name           = "group_auth_priv"
               security_level = "auth-priv"
               view           = "view_all"
            },
        ]
       id                          = (known after apply)
       location_of_device_variable = "{{snmp_location}}"
       name                        = "snmpv3"
       shutdown_variable           = "{{snmp_shutdown}}"
       trap_target_servers         = [
           {
               ip                        = "172.16.0.11"
               port                      = 514
               source_interface_variable = "{{snmp_trap_source_interface}}"
               user                      = "user01"
               vpn_id                    = 511
            },
        ]
       users                       = [
           {
               authentication_password = "<replace-password>"
               authentication_protocol = "sha"
               group                   = "group_auth_priv"
               name                    = "user01"
               privacy_password        = "<replace-password>"
               privacy_protocol        = "aes-cfb-128"
            },
        ]
       version                     = (known after apply)
       views                       = [
           {
               name = "view_all"
               oids = [
                   {
                       id = "1.3"
                    },
                   {
                       exclude = true
                       id      = "1.3.5"
                    },
                ]
            },
        ]
    }
```

Configure this resource using terraform and then go to UI -> edit -> save without any changes. Then do TF apply again. We expect no changes but terraform shows:
```
  # module.sdwan.sdwan_system_snmp_profile_parcel.system_snmp_profile_parcel["system1-snmp"] will be updated in-place
  ~ resource "sdwan_system_snmp_profile_parcel" "system_snmp_profile_parcel" {
        id                          = "87f104e7-cd65-4404-849c-db1b2ab0d309"
        name                        = "snmpv3"
      ~ trap_target_servers         = [
          ~ {
              + ip                        = "172.16.0.11"
              + port                      = 514
              + source_interface_variable = "{{snmp_trap_source_interface}}"
              + user                      = "user01"
              + vpn_id                    = 511
            },
        ]
      ~ version                     = 0 -> (known after apply)
        # (8 unchanged attributes hidden)
    }
```

**Comment by tzarski0** [comment]
Fixed by commit 429249e
